src/1AC760.c

```diff
@@ -193,27 +289,103 @@ ApiStatus GetActorLevel(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "1AC760", PartnerDamageEnemy);
+ApiStatus PartnerDamageEnemy(Evt* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Bytecode* args = script->ptrReadPos;
+    Actor* enemy = get_actor(script->owner1.enemy);
+    s32 temp_s4 = *args++;
+    s32 flags;
+    s32 damageResult;
+    u8 statusChance;
+
+    gBattleStatus.currentAttackElement = *args++;
+    gBattleStatus.currentAttackEventSuppression = *args++;
+    gBattleStatus.currentAttackStatus = *args++;
+    gBattleStatus.currentAttackDamage = evt_get_variable(script, *args++);
+    gBattleStatus.powerBounceCounter = 0;
+    flags = *args++;
+
+    if ((flags & 0x30) == 0x30) {
+        battleStatus->flags1 |= 0x10;
+        battleStatus->flags1 |= 0x20;
+    } else if (flags & 0x10) {
+        battleStatus->flags1 |= 0x10;
+        battleStatus->flags1 &= ~0x20;
+    } else if (flags & 0x20) {
+        battleStatus->flags1 &= ~0x10;
+        battleStatus->flags1 |= 0x20;
+    } else {
+        battleStatus->flags1 &= ~0x10;
+        battleStatus->flags1 &= ~0x20;
+    }
+
+    if (flags & 0x40) {
+        gBattleStatus.flags1 |= 0x40;
+    } else {
+        gBattleStatus.flags1 &= -0x41;
+    }
+
+    if (flags & 0x200) {
+        gBattleStatus.flags1 |= 0x200;
+    } else {
+        gBattleStatus.flags1 &= -0x201;
+    }
+
+    if (flags & 0x80) {
+        gBattleStatus.flags1 |= 0x80;
+    } else {
+        gBattleStatus.flags1 &= -0x81;
+    }
+
+    if (flags & 0x800) {
+        gBattleStatus.flags1 |= 0x800;
+    } else {
+        gBattleStatus.flags1 &= -0x801;
+    }
+
+    statusChance = battleStatus->currentAttackStatus;
+    battleStatus->currentTargetID = enemy->targetActorID;
+    battleStatus->currentTargetPart = enemy->targetPartIndex;
+    battleStatus->statusChance = statusChance;
+    
+    if (statusChance == 0xFF) {
+        battleStatus->statusChance = 0;
+    }
+
+    battleStatus->statusDuration = (battleStatus->currentAttackStatus & 0xF00) >> 8;
```

Comment:
Looks like some kind of multiply?

Reply:
Sadly not, this seems to be some kind of hacky case from s32 to u8 they did here, and doing it explicity doesn't work.